packages/babel-core/src/config/config-chain.js

```diff
@@ -168,6 +169,7 @@ export function buildRootChain(
       babelrc = validatedFile.options.babelrc;
     }
     if (babelrcRoots === undefined) {
+      babelrcRootsDirectory = validatedFile.dirname;
```

Comment:
Could you explain me how this PR fixes the issue?
If I understand it correctly, this line is the only change: if `babelrcRoots`  undefined, we use `validatedFile.dirname` instead of `context.cwd`, which is the default value of `babelrcRootsDirectory` (which is the same as the old behavior).
In the issue you said that it occurs when `babelrcRoots` is set to _something_, so I don't understand how this PR fixes it.

Reply:
> ```
> babelrcRoots: [
>    ".",
>]
> ```
>the expectation is that the . is relative to the config file, but currently we are calculating it relative to the working directory.

But doesn't this PR only changes the resolution behavior if `babelrcRoots === undefined`? If it is an array we still resolve relative to `context.cwd`, because we never go inside that `if`.

Reply:
Oh, the `if (babelrcRoots === undefined) {` check here is if it was `undefined` in the `babel.transform` arguments, not if it was `undefined` in the config file.

With the way things are set up now, we entirely ignore `babelrcRoots` in the config file if it is passed in the programmatic arguments.

Reply:
Oh ok :+1: